project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/Util.kt

```diff
@@ -86,3 +99,25 @@ fun ProjectItemLensDescriptor.getLens() = when (this) {
 
 suspend fun CompilationException.apply(transformations: List<ExceptionTransformation>, context: IJDDContext) =
     transformations.fold(this) { acc, it -> acc.apply(it, context) }
+
+fun Project.acceptOnAllKotlinFiles(visitor: KotlinRecursiveElementVisitor) {
+    val snapshotPath = service<MinimizationPluginSettings>().state.temporaryProjectLocation
+    VfsUtil.visitChildrenRecursively(guessProjectDir()!!, object : VirtualFileVisitor<Void>() {
```

Comment:
1. Why not just get psi from the project dir and use accept on it?
2. Let's try to extract the functionality of `FileTreeHierarchicalDDGenerator#findPossibleRoots` and reuse it here (and maybe save "roots" in the context to reuse them efficiently)